src/qos_client/src/yubikey.rs

```diff
@@ -350,21 +351,35 @@ fn generate_random_rfc5280_serial() -> [u8; 20] {
 	serial
 }
 
-#[test]
-fn test_rfc5280_serial_generation_success() {
-	use x509_cert::certificate::Rfc5280;
+#[cfg(test)]
+mod tests {
+	use crate::yubikey::generate_random_rfc5280_serial;
+	use std::time::Duration;
+	use x509_cert::{serial_number::SerialNumber, time::Validity};
 
-	// the generation behavior is non-deterministic by design, try it a few dozen times
-	const TEST_ITERATIONS: usize = 100;
+	#[test]
+	fn test_rfc5280_serial_generation_success() {
+		use x509_cert::certificate::Rfc5280;
 
-	for _ in 0..TEST_ITERATIONS {
-		let serial_data = generate_random_rfc5280_serial();
+		// the generation behavior is non-deterministic by design, try it a few dozen times
+		const TEST_ITERATIONS: usize = 100;
 
-		// ensure most significant bit is 0
-		assert_eq!(serial_data[0] & 0x80, 0);
+		for _ in 0..TEST_ITERATIONS {
+			let serial_data = generate_random_rfc5280_serial();
 
-		// test if serial conversion works
-		let _serial: SerialNumber<Rfc5280> =
-			SerialNumber::new(&serial_data).unwrap();
+			// ensure most significant bit is 0
+			assert_eq!(serial_data[0] & 0x80, 0);
+
+			// test if serial conversion works
+			let _serial: SerialNumber<Rfc5280> =
+				SerialNumber::new(&serial_data).unwrap();
+		}
+	}
+
+	#[test]
+	#[should_panic]
+	fn test_validity_calculation() {
+		let _validity =
+			Validity::from_now(Duration::from_secs(u64::MAX)).unwrap();
 	}
```

Comment:
Your test has been added for future reference. Feel free to tell if it is not worth including